chromium_src/chrome/browser/net/stub_resolver_config_reader.cc

```diff
@@ -0,0 +1,114 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/net/stub_resolver_config_reader.h"
+#include "brave/components/brave_vpn/buildflags/buildflags.h"
+#include "build/buildflag.h"
+#include "chrome/browser/net/secure_dns_config.h"
+#include "chrome/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/network_service_instance.h"
+#include "net/dns/public/dns_over_https_config.h"
+#include "services/network/public/mojom/network_service.mojom.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+#if BUILDFLAG(IS_WIN)
+namespace {
+
+bool ShouldOverride(PrefService* local_state,
+                    SecureDnsConfig::ManagementMode management_mode,
+                    bool is_managed) {
+  if (management_mode != SecureDnsConfig::ManagementMode::kNoOverride ||
+      is_managed) {
+    // there is already a managed policy or parental control in place
+    return false;
+  }
+  auto* pref = local_state->FindPreference(prefs::kBraveVpnDnsConfig);
+  if (!pref)
+    return false;
+  auto* value = pref->GetValue();
```

Comment:
checking to see if `kBraveVpnDnsConfig` is non-empty is sufficient here, but this also needs to check the doh mode because we shouldn't override if it's already `kSecure`

Reply:
added `kSecure` and it checks of `kBraveVpnDnsConfig` is empty only

Reply:
It's still checking the actual servers. This pref should either have a value or not

Reply:
asked in DM